src/pyhf/tensor/numpy_backend.py

```diff
@@ -654,4 +656,4 @@ def transpose(self, tensor_in: Tensor[T]) -> ArrayLike:
 
         .. versionadded:: 0.7.0
         """
-        return tensor_in.transpose()
+        return cast(ArrayLike, tensor_in.transpose())
```

Comment:
@kratsg this PR is mostly just a cherry-pick from your first commit in PR https://github.com/scikit-hep/pyhf/pull/2566, but to get Python 3.10 typing with `mypy` to pass I had to cast the output of `transpose` to `ArrayLike`.

I'm not sure if this makes sense or not, so I think you'll need to weigh in here.